walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,58 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.stream.Collectors;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.stream.Stream;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> possibleTimes = new ArrayList();
+    Stream.concat(events.stream()
+    	.filter(event -> areAnyAttendeeBusy(event, request))
+    	.map(Event::getWhen), Stream.of(TimeRange.fromStartDuration(TimeRange.END_OF_DAY+1, 0)))
+    .sorted(TimeRange.ORDER_BY_START)
+    .reduce(TimeRange.fromStartDuration(0, 0), (previous, current) -> 
+      mergeTimes(previous, current, request, possibleTimes)
+    );
```

Comment:
Formatting of Steam-based code can get a little unwieldy with larger pipelines.

I'd recommend something like this:

```java
Stream.concat(
    events.stream()
        .filter(event -> areAnyAttendeeBusy(event, request))
        .map(Event::getWhen),
    Stream.of(TimeRange.fromStartDuration(TimeRange.END_OF_DAY+1, 0)))
    .sorted(TimeRange.ORDER_BY_START)
    .reduce(
        TimeRange.fromStartDuration(0, 0),
        (previous, current) -> mergeTimes(previous, current, request, possibleTimes));
```

This follows the Google Java style guide's section on [indentation](https://google.github.io/styleguide/javaguide.html#s4.5.2-line-wrapping-indent) for line-wrapping (at least +4 spaces).

With this style, it's very easy to see what is being passed to `Stream.concat`: every +4 indent starts a new argument, and if that argument requires multiple lines they're indented further.

Once we get past the arguments, the lines starting with `.` on the later operations (`.sorted`, `.reduce`) make it clear we're now operating on the result of `Stream.concat`.

This is very much open to debate but I've found this approach to be pretty readable.